app/api-v1/controllers/Order/helpers.js

```diff
@@ -0,0 +1,40 @@
+const db = require('../../../db')
+const { NoTokenError, NothingToProcess } = require('../../../utils/errors')
+
+exports.mapOrderData = async (data) => {
+  if (!data.items || data.items.length < 1) throw new NothingToProcess()
+  const records = await db.getRecipeByIDs(data.items)
+  const tokenIds = records.map((el) => el.token_id)
+  if (!tokenIds.every(Boolean)) throw new NoTokenError('recipes')
+
+  const recipes = tokenIds.reduce((output, id) => {
+    if (id) {
+      output[id] = {
+        type: 'TOKEN_ID',
+        value: id,
+      }
+    }
+
+    return output
+  }, {})
+
+  return {
+    inputs: tokenIds,
```

Comment:
`outputs` will also need new copies of the burned `recipe` inputs so that each `recipe` can be used again by something else, like a different `order`. Without making new ones, a new `order` of the same recipes would fail because of [this check](https://github.com/digicatapult/dscp-node/blob/1deadf15a7ae6660ac6f1aedcfb0cc64f02fc39d/pallets/simple-nft/src/lib.rs#L202).

`recipe` output format:
```
| Roles                | Metadata                   |
| :------------------- | :------------------------- |
| Owner: `BAE`         | `<Literal>` type: `RECIPE` |
| Buyer: `BAE`         |                            |
| Supplier:`SupplierX` |                            |

Reply:
is enough to just add a type?

Reply:
I think so for now. We can look up the chain of the `recipe` token's parents for the other metadata